### PR TITLE
Virtual File System Explorer Module

### PR DESCRIPTION
**Pull request by liangshiw**
Resolve https://github.com/abpframework/abp/issues/1609

Provide a simple page to view all virtual files.

Use:

```csharp
[DependsOn(typeof(AbpVirtualFileExplorerWebModule))]
public class <YourProjectWebModule> : AbpModule
```

Options:

```csharp
public class AbpVirtualFileExplorerOptions
{
     /// <summary>
     /// Default: true.
     /// </summary>
     public bool IsEnabled { get; set; } = true;
}
```

Demo:

image

image

### CODE REVIEW

File: modules/virtual-file-explorer/src/Volo.Abp.VirtualFileExplorer.Web/FodyWeavers.xml [hunk 1]
```diff
@@ -0,0 +1,3 @@
+﻿<Weavers xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:noNamespaceSchemaLocation="FodyWeavers.xsd">
+  <ConfigureAwait />
```
Comment (maliming) [inline modules/virtual-file-explorer/src/Volo.Abp.VirtualFileExplorer.Web/FodyWeavers.xml:2]: <ConfigureAwait ContinueOnCapturedContext="false" />

Comment (liangshiw) [reply modules/virtual-file-explorer/src/Volo.Abp.VirtualFileExplorer.Web/FodyWeavers.xml:2]: Ok

**Comment by hikalkan** [comment]
That's a great module! The only thing is: You embedded primjs into the module. However, using a NPM package would be great. As a result, this module will have a NPM package (just like https://github.com/abpframework/abp/blob/dev/npm/packs/docs/package.json). Its package name can be "@abp/virtual-file-explorer" Once the final application adds this package and runs yarn & gulp, prismjs will be added and perfectly works. You just need to use the PrismJs contributors. 

@maliming can help you if you need?

**Comment by maliming** [comment]
This requires some work and testing. Move to v2.9.

**Comment by maliming** [comment]
@hikalkan We can release it in 2.8.

**Comment by kurtwilbies** [comment]
I dont like the naming, e.g. DemoApp.

**Comment by hikalkan** [comment]
Thanks a lot @liangshiw

**Pull request merged**